lib/ocpp/v201/charge_point.cpp

```diff
@@ -2952,5 +2947,65 @@ void ChargePoint::scheduled_check_v2g_certificate_expiration() {
             .value_or(12 * 60 * 60)));
 }
 
+void ChargePoint::set_operative_status(std::optional<int32_t> evse_id, std::optional<int32_t> connector_id,
+                                       OperationalStatusEnum new_status, bool persist) {
+    if (!evse_id.has_value()) {
+        // The whole CS is targeted
+        this->component_state_manager->set_cs_individual_operational_status(new_status, persist);
+        this->trigger_callbacks_if_effective_state_changed();
+    } else {
+        // An EVSE or Connector is targeted
+        this->evses.at(evse_id.value())->set_operative_status(connector_id, new_status, persist);
+    }
+}
+
+void ChargePoint::trigger_change_effective_availability_callback() {
+    if (this->callbacks.change_cs_effective_availability_callback.has_value()) {
+        this->callbacks.change_cs_effective_availability_callback.value()(
+            this->component_state_manager->get_cs_individual_operational_status());
+    }
+    for (auto& [evse_id, evse] : this->evses) {
+        if (evse != nullptr) {
+            evse->trigger_change_effective_availability_callback();
+        }
+    }
+}
+
+void ChargePoint::notify_user_if_all_connectors_inoperative() {
+    // Check that all connectors on all EVSEs are inoperative
+    for (auto& [evse_id, evse] : this->evses) {
+        if (evse != nullptr) {
+            for (int connector_id = 1; connector_id <= evse->get_number_of_connectors(); connector_id++) {
+                OperationalStatusEnum connector_status =
+                    this->component_state_manager->get_connector_effective_operational_status(evse_id, connector_id);
+                if (connector_status == OperationalStatusEnum::Operative) {
+                    return;
+                }
+            }
+        }
+    }
+    // Check succeeded, trigger the callback
+    if (this->callbacks.all_connectors_unavailable_callback.has_value()) {
+        this->callbacks.all_connectors_unavailable_callback.value()();
+    }
+}
```

Comment:
(you can just ignore this comment if you don't agree)

I'd rather have a 
```C++
bool ChargePoint::are_all_connectors_inoperative()
```
that just runs the first 21 lines of this method (early returns false else true)
and then at the places where this is currently called a 

```C++
  if (this->callbacks.all_connectors_unavailable_callback.has_value() && this->are_all_connectors_inoperative()) {
      this->callbacks.all_connectors_unavailable_callback.value()();
  }
```

Just for the reason I did not get what "notify user" means before looking into the definition of the method.